### PR TITLE
Server launchers: easily start, stop, show and deploy servers

### PR DESCRIPTION
**Pull request by giuliojiang**
**./RunServer [start|stop|show]** will start or stop all necessary server components (provided that they have been compiled correctly). The started process will run in a background daemon, using `screen` multiplexer. So screen must be installed on the PC. You can also type `./RunServer show` to check if there are servers running, or to attach to its standard console.

**./AutoDeploy <destination> <domain>** will automatically deploy the server and start it. Will be very useful for teamcity auto-deployment

**note** AutoDeploy doesn't work properly yet until RunServer will be available on Master branch

### CODE REVIEW

**Comment by mihaicar** [comment]
Are we still planning on integrating with labTS? If yes, we shouldnt forget to install screen on the vm, right?

**Comment by helicopter88** [comment]
We should keep a file with all the dependencies we need

**Comment by giuliojiang** [comment]
server distributions of Linux pretty much all have screen by default, because it's heavily used by many services.
As for dependencies and requirements, it's probably a good idea to keep track of them ;)

**Comment by mihaicar** [comment]
sounds good then

File: AutoDeploy [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import subprocess
+
+screen_name = 'alabnodaemonservice'
```
Comment (helicopter88) [inline AutoDeploy]: what about alabnod to make it sound more like a linux service

File: RunServer [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import os
+import sys
+import subprocess
+
+# Starts all the necessary server components as a background process
+# Also allows to reattach to the multiplexed terminal,
+# and to kill the process
+
+screen_name = 'alabnodaemonservice'
```
Comment (helicopter88) [inline RunServer]: same as above

File: RunServer [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+exec_dir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)))
+os.chdir(exec_dir)
+
+def start_server():
+    start_server_command = './infrastructure/runservers.sh'
+    cmd = 'screen -A -m -d -S {} {}'.format(screen_name, start_server_command)
```
Comment (helicopter88) [inline RunServer]: Maybe explain what those flags do in a comment?

File: RunServer [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+command = sys.argv[1]
+
+if command == 'show':
+    show_server()
+elif command == 'start':
+    start_server()
+elif command == 'stop':
+    kill_server()
+else:
+    exit_with_error()
```
Comment (helicopter88) [inline RunServer]: no new line at the end of the file

File: RunServer (hunk 3 above)
Comment (giuliojiang) [inline RunServer]: I have no idea...

File: AutoDeploy [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+def call_and_check(cmd):
+    code = subprocess.call(cmd, shell=True)
+    if code != 0:
+        print('Command <{}> returned code {}'.format(cmd, code))
+        sys.exit(1)
+    return
+
+# get output directory
+if len(sys.argv) != 3:
+    print('Usage: ./AutoDeploy <destination> <example.com>')
```
Comment (Proca-Consul) [inline AutoDeploy]: Very minor thing: maybe 'Usage: ./AutoDeploy <destination> <domain>' is more consistent? :)

**Pull request merged**